pkg/volume/projected/projected.go

```diff
@@ -321,12 +321,13 @@ func (s *projectedVolumeMounter) collectData(mounterArgs volume.MounterArgs) (ma
 		case source.ServiceAccountToken != nil:
 			tp := source.ServiceAccountToken
 
-			// When FsGroup is set, we depend on SetVolumeOwnership to
-			// change from 0600 to 0640.
 			mode := *s.source.DefaultMode
-			if mounterArgs.FsUser != nil || mounterArgs.FsGroup != nil {
+			if mounterArgs.FsUser != nil {
```

Comment:
This block is confusing - what if both are set?  Can we leve some comments, please

Reply:
I've clarified the block a little: FsGroup shall take precedence (so outcome still follows [Case 1 here](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/1205-bound-service-account-tokens#proposed-heuristics)).